### PR TITLE
Add subcode to ESWrapper

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Add subcode to ESWrapper

### Other changes

 - change `errorName` from  `domain-subdomain-error` to  `domain.subdomain.error` (eg: `errorName: 'external.elasticsearch.index_or_collection_does_not_exist'`)

### CODE REVIEW

File: lib/util/esWrapper.js [hunk 1]
```diff
@@ -114,63 +121,106 @@ class ESWrapper {
...
 
     let message = error.message || '';
 
-    const messageReplaced = errorMessagesMapping.some(mapping => {
-      message = message.replace(mapping.regex, mapping.replacement);
-      return message !== error.message;
-    });
+    let kuzzleError = null;
+
+    for (const mapping of errorMessagesMapping) {
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: MAJOR Reduce the total number of "break" and "continue" statements in this loop to use one at most. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ATooManyBreakOrContinueInLoop)

File: lib/util/esWrapper.js [hunk 2]
```diff
@@ -114,63 +121,106 @@ class ESWrapper {
...
+      const matches = message.match(mapping.regex, mapping.replacement);
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: CRITICAL "match" expects 1 argument, but 2 were provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)

File: lib/util/esWrapper.js [hunk 3]
```diff
@@ -114,63 +121,106 @@ class ESWrapper {
...
-          debug('unhandled "NotFound" elasticsearch error: %a', error);
+        if (error.body && error.body.error) {
+          message = error.body.error
+            ? `${error.body.error.reason}: ${error.body.error['resource.id']}`
+            : `${error.message}: ${error.body._id}`;
         }
 
-        kuzzleError = new NotFoundError(error);
+        kuzzleError = errorsManager.getError(
+          'external',
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: MAJOR Remove this useless assignment to local variable "message" [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1854)

File: lib/util/esWrapper.js [hunk 4]
```diff
@@ -114,63 +121,106 @@ class ESWrapper {
...
-      message = message.replace(mapping.regex, mapping.replacement);
-      return message !== error.message;
-    });
+    let kuzzleError = null;
+
+    for (const mapping of errorMessagesMapping) {
+      const matches = message.match(mapping.regex, mapping.replacement);
+
+      if (!matches) {
+        continue;
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: MINOR Remove this "continue" statement. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AContinueStatement)

**Comment by kuzzle** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [throw.js#](https://github.com/kuzzleio/kuzzle/blob/978b5ecd7c1e37add44041a681c2279bbd0aab2d/lib/config/error-codes/throw.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [throw.js#](https://github.com/kuzzleio/kuzzle/blob/978b5ecd7c1e37add44041a681c2279bbd0aab2d/lib/config/error-codes/throw.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [throw.js#](https://github.com/kuzzleio/kuzzle/blob/fc36ba6e98c0753ad3b0e91366503bf61dc8325d/lib/config/error-codes/throw.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: lib/config/error-codes/throw.js [hunk 5]
```diff
@@ -24,42 +24,73 @@ const { format } = require('util'),
...
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ */
+function wrap (domain, subdomain) {
+  return {
+    getError: (error, ...placeholders) => getError(domain, subdomain, error, ...placeholders),
+    throw: (error, ...placeholders) => throwError(domain, subdomain, error, ...placeholders)
+  };
+}
+
+module.exports = {
```
Comment (kuzzle) [inline lib/config/error-codes/throw.js]: MINOR Group all shorthand properties at either the beginning or end of this object declaration. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)

**Comment by kuzzle** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [throw.js#](https://github.com/kuzzleio/kuzzle/blob/b62871c2cbb51facefc385d81a9730fb7529f6f4/lib/config/error-codes/throw.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [throw.js#](https://github.com/kuzzleio/kuzzle/blob/26cf2993d994b62f5af56b9dd33d710ad540a517/lib/config/error-codes/throw.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [throw.js#](https://github.com/kuzzleio/kuzzle/blob/a21048303ec8dc21618407694ef4bfbea9ba32e9/lib/config/error-codes/throw.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Aschen** [comment]
> You must remove `package-lock.json` from updated files.
> There is also something weird about changes on Auth Local plugin.

@alexandrebouthinon fixed

**Pull request merged**